src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp

```diff
@@ -547,40 +547,74 @@ struct ONNXGenericOpToCall : public mlir::OpConversionPattern<OP_TYPE> {
             typeConverter, ctx, /*benefit higher than default*/ 10),
         opsForCall(opsForCall) {}
   std::string opsForCall;
+  
+  mlir::LogicalResult matchAndRewrite(OP_TYPE onnxOp,
+      ADAPTOR_TYPE adaptor,
+      mlir::ConversionPatternRewriter &rewriter) const final {
 
-  mlir::LogicalResult match(OP_TYPE onnxOp) const final {
-    mlir::Operation *op = onnxOp.getOperation();
-    if (!checkOpToCall(op, opsForCall))
+       mlir::Operation *op = onnxOp.getOperation();
+       if (!checkOpToCall(op, opsForCall))
       return mlir::failure();
 
-    // Additional checks
-
-    return mlir::success();
-  }
-  void rewrite(OP_TYPE onnxOp, ADAPTOR_TYPE adaptor,
-      mlir::ConversionPatternRewriter &rewriter) const final {
-    mlir::Operation *op = onnxOp.getOperation();
-    mlir::Location loc = onnx_mlir::ONNXLoc<OP_TYPE>(op);
-    mlir::ValueRange operands = adaptor.getOperands();
-
-    // Get shape.
-    MultiDialectBuilder<IndexExprBuilderForKrnl, MemRefBuilder> create(
-        rewriter, loc);
+    // Additional checks TODO: Should there be a conditional check for write?
```

Comment:
@Sunny-Anand what is done here and elsewhere is that the separate `match` and `rewrite` are now combined, and you just pasted the two together, is that correct?